tools/accuracy_checker/openvino/tools/accuracy_checker/evaluators/custom_evaluators/opennmt_encoder_decoder_generator_evaluator.py

```diff
@@ -0,0 +1,473 @@
+        super().__init__(network_info, launcher)
+        stages = ['encoder', 'decoder', 'generator']
+        if models_args and not delayed_model_loading:
+            for idx, stage in enumerate(stages):
+                stage_info = network_info.get(stage, {})
+                if not contains_any(stage_info, ['model', 'onnx_model']) and models_args:
+                    stage_info['model'] = models_args[idx if len(models_args) > idx else 0]
+                    stage_info['_model_is_blob'] = is_blob
+                network_info.update({stage: stage_info})
+
+        if not contains_all(network_info, stages) and not delayed_model_loading:
+            raise ConfigError('network_info should contain encoder, decoder and generator fields')
+
+        self._encoder_mapping = {
+                'dlsdk': EncoderDLSDKModel,
+                'onnx_runtime': EncoderONNXModel,
+            }
+        self._decoder_mapping = {
+                'dlsdk': DecoderDLSDKModel,
+                'onnx_runtime': DecoderONNXModel,
+            }
+
+        self._generator_mapping = {
+                'dlsdk': GeneratorDLSDKModel,
+                'onnx_runtime': GeneratorONNXModel
+            }
+
+        self.encoder = create_model(network_info['encoder'], launcher, self._encoder_mapping, 'encoder',
+                                    delayed_model_loading)
+        self.decoder = create_model(network_info['decoder'], launcher, self._decoder_mapping, 'decoder',
+                                    delayed_model_loading)
+        self.generator = create_model(network_info['generator'], launcher, self._generator_mapping, 'generator',
+                                      delayed_model_loading)
+
+        self.store_encoder_predictions = network_info['encoder'].get('store_predictions', False)
+        self.store_decoder_predictions = network_info['decoder'].get('store_predictions', False)
+        self._encoder_predictions = [] if self.store_encoder_predictions else None
+        self._decoder_predictions = [] if self.store_decoder_predictions else None
+
+        self._part_by_name = {'encoder': self.encoder, 'decoder': self.decoder, 'generator': self.generator}
+        self._raw_outs = OrderedDict()
+
+        self.adapter = create_adapter(launcher.config.get('adapter', 'nmt'))
+
+        self.processing_frames_buffer = None
+
+    def predict(self, identifiers, input_data, encoder_callback=None):
+        predictions, raw_outputs = [], []
+
+        for data in input_data:
+            decode_strategy = BeamSearch(self.decoder.network_info)
+
+            src_len = np.array([len(data)])
+            h, c, memory = self.encoder.predict(identifiers, {'src': np.array([[[t]] for t in data]),
+                                                              'src_len': src_len})
+            if encoder_callback:
+                encoder_callback((h, c, memory, src_len))
```

Comment:
not sure that you use this callback properly
it is callback for providing  opportunity to calculate statistics in POT, it requires to provide full dictionary of outputs, not only meaningful for pipeline (pot can mark additional layers as outputs for getting their calculation results

Reply:
fixed